x-pack/docs/en/rest-api/security/update-api-key.asciidoc

```diff
@@ -1,5 +1,277 @@
 [role="xpack"]
 [[security-api-update-api-key]]
-=== Update API key information API
+=== Update API key API
 
-coming::[8.4.0]
+++++
+<titleabbrev>Update API key</titleabbrev>
+++++
+
+[[security-api-update-api-key-request]]
+==== {api-request-title}
+
+`PUT /_security/api_key/<id>`
+
+[[security-api-update-api-key-prereqs]]
+==== {api-prereq-title}
+
+* To use this API, you must have at least the `manage_own_api_key` cluster privilege.
+Users can only update API keys that they created or that were granted to them.
+To update another user's API key, use the <<run-as-privilege,`run_as` feature>>
+to submit a request on behalf of another user.
+
+IMPORTANT: It's not possible to use an API key as the authentication credential for this API.
+To update an API key, the owner user's credentials are required.
+
+[[security-api-update-api-key-desc]]
+==== {api-description-title}
+
+Use this API to update API keys created by the <<security-api-create-api-key,create API Key>> or <<security-api-grant-api-key,grant API Key>> APIs.
+It's not possible to update expired API keys, or API keys that have been invalidated by <<security-api-invalidate-api-key,invalidate API Key>>.
+
+This API supports updates to an API key's access scope, i.e., its permissions, and its metadata.
+The access scope of an API key is derived from the <<security-api-update-api-key-api-key-role-descriptors,`role_descriptors`>> you specify in the request, and a snapshot of the owner user's permissions at the time of the request.
+The snapshot of the owner's permissions is updated automatically on every call.
+
+[IMPORTANT]
+====
+A call to this API might change the API key's access scope, even if you don't specify new
+<<security-api-update-api-key-api-key-role-descriptors,`role_descriptors`>> in the request.
+This behavior can occur if the owner user's permissions have changed since the last modification of the API key.
```

Comment:
My attempt to cover both creation and update:
```suggestion
It can occur if the owner user's permissions have changed and are different from the snapshot version stored as part of the API key.
```

Reply:
What about "have changed since the creation or last modification of the API key"? It keeps things more high level, without going into detail about the snapshot.

Alternatively:

"can occur if the snapshot of the owner permissions on the API key is stale, i.e., the owner permissions have since diverged (changed?)."

Reply:
What if we flipped this statement to move the conditional part about `role_descriptors` to the end? I think that moving the explanation of why this can occur helps to clarify the initial sentence:

>A call to this API might change the API key's access scope, which can occur if the owner user's
>permissions have changed since the API key was created or last modified. This change can occur
>even if you don't specify new <<security-api-update-api-key-api-key-role-descriptors,`role_descriptors`>> in the request.

Reply:
I like it. Thanks!

Reply:
Discussed on slack, and going with a different tweak which keeps the callout specific to the "access change despite omitted role descriptors" issue.